src/import/load-costume.js

```diff
@@ -82,21 +52,85 @@ const loadCostumeFromAsset = function (costume, costumeAsset, runtime, optVersio
         };
         imageElement.addEventListener('error', onError);
         imageElement.addEventListener('load', onLoad);
-        imageElement.src = costumeAsset.encodeDataURI();
+        let src = costumeAsset.encodeDataURI();
+        if (costume.bitmapResolution === 1 && !runtime.v2BitmapAdapter) {
+            log.error('No V2 bitmap adapter present; bitmaps may not render correctly.');
+        } else if (costume.bitmapResolution === 1) {
+            src = runtime.v2BitmapAdapter.convertResolution1Bitmap(src, dataURI => {
+                if (dataURI) {
+                    // Put back into storage
+                    const storage = runtime.storage;
+                    costume.assetId = storage.builtinHelper.cache(
+                        storage.AssetType.ImageBitmap,
+                        storage.DataFormat.PNG,
+                        runtime.v2BitmapAdapter.convertDataURIToBinary(dataURI)
```

Comment:
Any reason why convertResolution1Bitmap doesn't call convertDataURIToBinary itself?

Reply:
Here the VM needs both the dataURI (to set as the image src) and the Uint8Array (to put back in storage) so I think it makes sense to return the dataURI and let VM convert it to a Uint8Array when it wants.